test-refined/shared/src/main/scala/zio/test/refined/NumericInstances.scala

```diff
@@ -0,0 +1,63 @@
+package zio.test.refined
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.internal.WitnessAs
+import eu.timepit.refined.numeric.{Greater, Less}
+import zio.test.Gen
+import zio.test.magnolia.DeriveGen
+
+object numeric extends NumericInstances
+
+trait NumericInstances {
+
+  def intGreaterThan[N](implicit
+    wn: WitnessAs[N, Int]
+  ): DeriveGen[Int Refined Greater[N]] =
+    DeriveGen.instance(Gen.int(wn.snd, Int.MaxValue).map(Refined.unsafeApply))
+
+  def longGreaterThan[N](implicit
+    wn: WitnessAs[N, Long]
+  ): DeriveGen[Long Refined Greater[N]] =
```

Comment:
Minor:
Let's do `Refined[Long, Greater[N]]` to keep the consistency, because rest of the PR is in that format. 
That said, I am happy to change rest of the PR to this style if you need `Refined` as infix.